smartsim/entity/model.py

```diff
@@ -315,6 +333,30 @@ def _set_colocated_db_settings(
 
         self.run_settings.colocated_db_settings = colo_db_config
 
+    @staticmethod
+    def _create_pinning_string(
+        pin_ids: t.Optional[t.Iterable[t.Union(int, t.Iterable[int])]],
+        cpus: int
+        ):
+        """Create a comma-separated string CPU ids. By default, None returns
+        0,1,...,cpus-1; an empty iterable will disable pinning altogether,
+        and an iterable constructs a comma separate string (e.g. 0,2,5)
+        """
+        if pin_ids is None:
+            return ','.join(str(i) for i in range(cpus))
+        elif not pin_ids:
+            return None
+        elif isinstance(pin_ids, collections.abc.Iterable):
+            pin_list = []
```

Comment:
Thoughts on adding an
```suggestion
        elif isinstance(pin_ids, collections.abc.Iterable):
            if "darwin" in sys.platform:
                raise ValueError("Custom DB CPU pinning not available on darwin")
            pin_list = []
```
block here?

Reply:
Changed it so that it warns if pin_ids isn't an empty list on MacOS. It then follows the normal logic below.